src/build/ci/check-module-package-conventions.sh

```diff
@@ -0,0 +1,130 @@
+#! /usr/bin/env bash
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+# The purpose of this ci script is to ensure that a pull request
+# doesn't unintentionally add a jar resource that would cause
+# problems with jar sealing by breaking our package naming
+# conventions, which are to use package names based on the
+# module, so they are unique.
+NUM_EXPECTED=0
+ALLOWED=(
```

Comment:
why not just allow everything in src/main/resources and src/test/resources? Do we need to enumerate here?

Reply:
We need to enumerate because of the way I wrote the script's grep behavior... copied from other similar CI scripts. Allowing wildcard or pattern parsing gets very tricky.

Most of these should get cleaned up with #5867 , so I think it's better to have them enumerated for now.